astropy/coordinates/sky_coordinate.py

```diff
@@ -227,7 +230,7 @@ def _parse_inputs(self, args, kwargs):
                 # coord_kwargs will contain keys like 'ra', 'dec', 'distance'
                 # along with any frame attributes like equinox or obstime which
                 # were explicitly specified in the coordinate object (i.e. non-default).
-                coord_kwargs = _parse_coordinate_arg(args[0], frame, units)
+                coord_kwargs = _parse_coordinate_arg(args[0], frame, units, kwargs)
```

Comment:
@eteq- I'm not sure we're on the same page here.  I'm looking at the code and on line 228 there is the statement:
```
if kwargs:
    raise ValueError(...)
```
Then on line 240 there is the call `coord_kwargs = _parse_coordinate_arg(args[0], frame, units, kwargs)`, which must be the same as `coord_kwargs = _parse_coordinate_arg(args[0], frame, units, {})`.  

I haven't traced through the logic but I was suggesting `coord_kwargs = _parse_coordinate_arg(args[0], frame, units, valid_kwargs)`.   My guess is that it would allow something like:
```
sc = SkyCoord('1d', '2d')
sc2 = SkyCoord([sc, sc], obstime='J2001.12')
```
where `obstime` becomes a `valid_kwarg`.